### PR TITLE
Implement as class and port tests to Brittle

### PR DESCRIPTION
**Pull request by kasperisager**
Closes #22

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,176 +1,179 @@
...
+const constants = fs.constants || require('constants') // eslint-disable-line n/no-deprecated-api
 
-inherits(RandomAccessFile, RandomAccess)
+const READONLY = constants.O_RDONLY
+const READWRITE = constants.O_RDWR | constants.O_CREAT
 
-RandomAccessFile.prototype._open = function (req) {
-  var self = this
+module.exports = function randomAccessFile (filename, opts) {
+  return new RandomAccessFile(filename, opts)
```
Comment (mafintosh) [inline index.js:11]: does this pattern work if someone did new RandomAccessFile from the outside?

Comment (kasperisager) [reply index.js:11]: Yep:

```
> const RAF = require('.')
undefined
> new RAF('foo.txt')
RandomAccessFile {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  _queued: [],
  _pending: 0,
  _needsOpen: true,
  opened: false,
  closed: false,
  destroyed: false,
  preferReadonly: true,
  readable: true,
  writable: true,
  deletable: true,
  statable: true,
  directory: null,
  filename: 'foo.txt',
  fd: 0,
  _size: 0,
  _truncate: false,
  _rmdir: false,
  _lock: [Function: noLock],
  _sparse: [Function: noLock],
  _alloc: [Function: allocUnsafe],
  [Symbol(kCapture)]: false
}
```

Comment (mafintosh) [reply index.js:10]: +1 then, been thinking if we should just eat the complexity and move to this always being a class directly btw

Comment (kasperisager) [reply index.js:10]: That does break the reasonably nice way of passing storage constructor functions as arguments:

```
const core = new Hypercore(RAF)
```

Comment (juliangruber) [reply index.js]: or we simplify the api and have either all functions or all classes (and always call `new`)

```js
const core = new Hypercore(filename => new RAF(filename))
```

```js
const core = new Hypercore(RAF)
```

File: test.js [hunk 2]
```diff
@@ -1,247 +1,268 @@
-var raf = require('./')
-var tape = require('tape')
-var os = require('os')
-var path = require('path')
-var fs = require('fs')
-var mkdirp = require('mkdirp')
-var isWin = process.platform === 'win32'
+const test = require('brittle')
```
Comment (juliangruber) [inline test.js:1]: what is the motivation for moving from `tape` to `brittle`? Maybe we could consider the new experimental test runner from node core https://github.com/nodejs/node-core-test, in hope that we'll eventually consume it from stdlib?

Comment (kasperisager) [reply test.js:1]: Brittle is overall solid and the rest of the ecosystem, including Hypercore and Hyperswarm, have mostly moved to Brittle as well.

Comment (juliangruber) [reply test.js:1]: Thanks!

File: index.js (hunk 1 above)
Comment (kasperisager) [reply index.js]: After discussing it with @mafintosh, that's the direction we'll go with, yes. It felt like the approach with the fewest surprises.

File: test.js (hunk 2 above)
Comment (kasperisager) [reply test.js:1]: Pleasure!

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

File: index.js [hunk 3]
```diff
@@ -1,176 +1,172 @@
...
+const READWRITE = constants.O_RDWR | constants.O_CREAT
 
-RandomAccessFile.prototype._open = function (req) {
-  var self = this
+module.exports = class RandomAccessFile extends RandomAccessStorage {
+  constructor (filename, opts = {}) {
+    super()
 
-  mkdirp(path.dirname(this.filename), ondir)
+    if (opts.directory) filename = path.join(opts.directory, path.resolve('/', filename).replace(/^\w+:\\/, ''))
```
Comment (mafintosh) [inline index.js:13]: what does the regex do?

Comment (kasperisager) [reply index.js:13]: Looks like it removes the protocol portion if passing a URL. It wasn't changed from `master`: https://github.com/random-access-storage/random-access-file/blob/58d9e1aa3fe8cc3868451a299ae272d1fd983378/index.js#L18

**Pull request merged**